File: C/Program24.c
////////////////////////////////////////////////////////////////////////////////////
//                                                                                //
//  File name :     Program24.c                                                   //
//  Description :   Used to count digits from given number                        //
//  Author :        Pallavi Omprakash Malewar                                     //
//  Date :          19/07/2025                                                    //
//                                                                                //
////////////////////////////////////////////////////////////////////////////////////


#include<stdio.h>

int CountDigits(int iNumber)
{
    int iCount = 0;

    while( iNumber != 0 )
    {
        iNumber = iNumber / 10;
        iCount++;
    }
    
    return iCount;
}


int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number : ");
    scanf("%d" , &iValue);

    iRet = CountDigits(iValue);

    printf("Number of digits are : %d\n" , iRet);

    return 0;

}

File: JAVA/Program24.java
//////////////////////////////////////////////////////////////////////////////////////////
//                                                                                      //
//  File name :     Progra24.java                                                      //
//  Description :   Used to Print Given Pattern                                         //
//  Author :        Pallavi Omprakash Malewar                                           //
//  Date :          25/08/2025                                                          //
//                                                                                      //
//////////////////////////////////////////////////////////////////////////////////////////


// Input : 4
// Output : 4   *   3   *   2   *   1   *



import java.util.Scanner;

class Pattern
{
    public void Display(int iNo)
    {
        int iCount = 0;

        if(iNo < 0)
        {
            iNo = -iNo;     // Filter
        }

        for(iCount = iNo ; iCount >= 1 ; iCount--)
        {
            System.out.print(iCount + "\t*\t");
        }
        System.out.println();
    }
}

class Program24
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.print("Enter Number : ");
        iValue = sobj.nextInt();

        Pattern pobj = new Pattern();

        pobj.Display(iValue);

    }
}

// Time Complexity O(N)
// N is a Natural Number
// Where N >= 0 & N <= Infinity